### PR TITLE
Tweak run assignment for low throughput runs.

### PR DESCRIPTION
**Pull request by vdbergh**
We fix an issue created by #2443.

Assume that X is a low throughput test, currently with 0 cores. Then in current master it moves to the front of the queue. Assume it manages to pick up a large core worker Y. Since X is low throughput it will not pick up more workers. When Y finishes its task on X, X falls back to 0 cores and moves to the front of the queue again. Since Y is now looking for a new task it will probably pick up X again. So X and Y are permanently paired which is not desirable.

In this PR we fix this by replacing `run["cores"]` by
```
adjusted_cores = run["cores"] + (max_threads if run_id == last_run_id else 0)
```
In other works we are re-adding the numbers of cores that were freed by this worker when the previous task on this run finished.

### CODE REVIEW

File: server/fishtest/rundb.py [hunk 1]
```diff
@@ -1108,21 +1108,22 @@ def sync_request_task(self, worker_info):
         last_run_id = self.worker_runs.get(my_name, {}).get("last_run", None)
 
         def priority(run):
-            # A penalty term for working on the same run again. This amounts to adding the number
-            # of cores that were freed by this worker when the previous task finished.
-            # The added 1/2 * max_threads is to mitigate granularity issues with large core workers.
-            repeat_penalty = (
-                3 / 2 if str(run["_id"]) == last_run_id else 1 / 2
-            ) * max_threads
+            # We readd the number of cores that were freed by this worker when the previous
```
[the remark this answers is not in the document]
Comment (ppigazzini) [reply server/fishtest/rundb.py]: True!
On the fly I misread "we are readding" as "we are reading" :)

**Comment by ppigazzini** [comment]
DEV updated, workers joined.

File: server/fishtest/rundb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ppigazzini) [reply server/fishtest/rundb.py]: Fixed.

**Comment by ppigazzini** [comment]
I stopped my x86-64-avx512icl worker to leave that arch available to you.

**Comment by vdbergh** [comment]
Last push corrects a typo in the commit message.

**Comment by vdbergh** [comment]
It seems to working. I lowered the avx512icl test to low throughput and it is still getting cores but not immediately.
I am away for the day now so I shutting down my laptop with my 3 workers.

**Comment by ppigazzini** [comment]
Ok, enjoy your day!

**Comment by ppigazzini** [comment]
The "x86-64-sse41-popcnt" test is not trapping the "x86-64-sse41-popcnt" worker.

**Comment by vdbergh** [comment]
> The "x86-64-sse41-popcnt" test is not trapping the "x86-64-sse41-popcnt" worker.

It will not trap it if it can get the required throughput without it. Also a low throughput test may sometimes hijack a worker since a zero core test moves to the front of the queue.

I am thinking about a decaying average implementation which could fix this but it will take a bit of time.
